### PR TITLE
code: go errors 2nd attempt

### PR DESCRIPTION
**Pull request by RemiMattheyDoret**
## Context

This PR contains the exact same changes than in https://github.com/edgelaboratories/daycount/pull/62 plus a fix for access to go-errors private repo. I decided to open a new PR merging directly in main to clean up a bit the commit history of https://github.com/edgelaboratories/daycount/pull/62.

Similar to https://github.com/edgelaboratories/picsou/pull/330

This PR migrate daycount into using go-errors.

## Content

- Use go-error
- Exclude go-error from wrapcheck (linter)'s sight
- Adapt github actions to access private repo as dependency
   - Is it correct how it is done?
   - I hesitated to define a github action workflow template to avoid duplicating functionality but it felt unnecessary and not the point of this PR.

### CODE REVIEW

File: .github/workflows/golangci.yml [hunk 1]
```diff
@@ -7,7 +7,15 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+          persist-credentials: false
+          ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
```
Comment (olivierjacob) [inline .github/workflows/golangci.yml:14]: What exactly is the goal of this configuration?

**Review by borisdavid — APPROVED** [review]

File: .github/workflows/golangci.yml (hunk 1 above)
Comment (olivierjacob) [reply .github/workflows/golangci.yml:14]: My understanding of the action definition here is that when opening a PR, we're going to checkout the repository. Telling the action that the `ref` is `main` will force the checkout of the `main` branch, and the branch that we would like to build for the open PR.
Am I mistaken ?

Also why do we need access to another repo ? Shouldn't we just use a version of the package ?

Comment (RemiMattheyDoret) [reply .github/workflows/golangci.yml:14]: > Also why do we need access to another repo ? Shouldn't we just use a version of the package ?

I am not sure I understand that. My understanding is that the github action needs to download a private repo and it therefore needs the appropriate credentials for that.

Comment (RemiMattheyDoret) [reply .github/workflows/golangci.yml:14]: > Also why do we need access to another repo ? Shouldn't we just use a version of the package ?

I am not sure I understand that. My understanding is that the github action needs to download a private repo and it therefore needs the appropriate credentials for that.

> Telling the action that the ref is main will force the checkout of the main branch, and the branch that we would like to build for the open PR.

Yeah... I kinda copy-pasted things there. I am not sure of what I did. Would you suggest to remove the `ref` key or to set it to [.github.ref](https://docs.github.com/en/actions/learn-github-actions/variables#:~:text=processing%20using%20the-,%24%7B%7B%20github.ref%20%7D%7D,-context%20property.)? I'll try things out [there](https://github.com/edgelaboratories/daycount/pull/65)

Comment (RemiMattheyDoret) [reply .github/workflows/golangci.yml:14]: As discussed with @olivierjacob, it seems that daycount does to have access to the saem secrets than other repos. I'll clean up the history here and will create a new PR for fixing the secrets and will ping the SREs on that other PR.

**Comment by borisdavid** [comment]
Closing this one as we will redo it in go-libraries anyway

**Pull request closed without merging**